src/devices/Ssd1351/Ssd1351.cs

```diff
@@ -0,0 +1,275 @@
+{
+    /// <summary>
+    /// A single-chip CMOS OLED/PLED driver with controller for organic/polymer
+    /// light emitting diode dot-matrix graphic display system. 
+    /// </summary>
+    public partial class Ssd1351 : IDisposable
+    {
+
+        private readonly int _dcPinId;
+        private readonly int _resetPinId;
+        private readonly int _spiBufferSize;
+        private readonly bool _disposeGpioController = false;
+
+        private SpiDevice _spiDevice;
+        private IGpioController _gpioDevice;
+        private ColorDepth _colorDepth;
+        private ColorSequence _colorSequence;
+
+        private const byte SCREEN_WIDTH_PX = 128;
+        private const byte SCREEN_HEIGHT_PX = 128;
+        private const int DefaultSPIBufferSize = 0x1000;
+
+        /// <summary>
+        /// Initializes new instance of Ssd1351 device that will communicate using SPI bus.
+        /// A single-chip CMOS OLED/PLED driver with controller for organic/polymer
+        /// light emitting diode dot-matrix graphic display system. 
+        /// </summary>
+        /// <param name="spiDevice">The SPI device used for communication.</param>
+        /// <param name="gpioController">The GPIO controller used for communication and controls the the <paramref name="resetPin"/> and the <paramref name="dataCommandPin"/>.</param>
+        /// <param name="dataCommandPin">The id of the GPIO pin used to control the DC line (data/command).</param>
+        /// <param name="resetPin">The id of the GPIO pin used to control the /RESET line (data/command).</param>
+        /// <param name="spiBufferSize">The size of the SPI buffer. If data larger than the buffer is sent then it is split up into multiple transmissions. The default value is 4096.</param>
+        public Ssd1351(SpiDevice spiDevice, int dataCommandPin, int resetPin, int spiBufferSize = DefaultSPIBufferSize, IGpioController gpioController = null)
+        {
+            if (!InRange((uint)spiBufferSize, 0x1000, 0x10000))
+            {
+                throw new ArgumentException($"SPI Buffer Size must be between 4096 and 65536.", nameof(spiBufferSize));
+            }
+
+            _gpioDevice = gpioController;
+
+            _spiDevice = spiDevice ?? throw new ArgumentNullException(nameof(spiDevice));
+
+
+            _dcPinId = dataCommandPin;
+            _resetPinId = resetPin;
+
+            if(_gpioDevice == null)
+            {
+                _gpioDevice = new GpioController();
+                _disposeGpioController = true;
+            }
+
+            _gpioDevice.OpenPin(_dcPinId, PinMode.Output);
+            _gpioDevice.OpenPin(_resetPinId, PinMode.Output);
+
+            _spiBufferSize = spiBufferSize;
+
+            SetSegmentReMapColorDepth();
+        }
+
+        /// <summary>
+        /// Convert a color structure to a byte tuple represening the colour in 565 format.
+        /// </summary>
+        /// <param name="color">The color to be converted.</param>
```

Comment:
NIT: Do you mind adding the `<returns>`  comment to explain which order are the bytes returned as? I just ask this because I know that other bindings will want to reuse this method.